packages/core/src/sync-historical/service.ts

```diff
@@ -137,6 +140,7 @@ export class HistoricalSyncService extends Emittery<HistoricalSyncEvents> {
   private blockTasksEnqueuedCheckpoint = 0;
 
   private queue: Queue<HistoricalSyncTask>;
+  private debouncedEmitCheckpoint: (checkpoint: Checkpoint) => void;
```

Comment:
Any reason to define it here rather than as a private method near the bottom?

Reply:
Debounce returns a callback which has extra context about the previous invocations of that callback. You can't just call `debounce()` in a helper function, because then it doesn't have any information about when it was called last and what timers are set.